todo_app/app.py

```diff
@@ -21,39 +21,45 @@ def index():
 def render_cards(id):
     status = False
     cardsList = get_cards(id)
-    print(cardsList)
+    cardsList.sort(key=lambda x: x.status)
+    boardLists = get_boardLists(id)
 
-    return render_template("index.html", cardsList=cardsList, status=status)
+    return render_template(
+        "index.html",
+        cardsList=cardsList,
+        status=status,
+        boardLists=boardLists,
+        boardId=id,
+    )
 
 
-@app.route("/", methods=["POST"])
+@app.route("/add", methods=["POST"])
 def add_new_item():
-    item = request.form["item"]
-    add_item(item)
-    return redirect("/")
+    boardId = request.form.get("board_id")
+    inputItem = request.form.get("inputItem")
+    selectedList = request.form.getlist("selectedList")[0]
+    itemDescription = request.form.get("itemDescription")
+    dueDate = request.form.get("dueDate")
+    date_object = datetime.strptime(dueDate, "%d-%m-%y").date()
+    if inputItem:
+        add_card(selectedList, inputItem, itemDescription, date_object)
 
+    return redirect(f"/{boardId}")
 
-@app.route("/check", methods=["POST"])
+
+@app.route("/update", methods=["POST"])
 def update_item():
-    data = request.get_json()
-    item = get_item(data["id"])
-    item["status"] = "Completed"
-    save_item(item)
-    response = {"message": "Item updated successfully", "item": item}
-    return jsonify(response), 200
+    boardId = request.form.get("board_id")
```

Comment:
Nothing wrong with this, but having the user pass the board id through in every request just so we can redirect them feels a little odd - maybe one where we could stick the board id in their session cookie (where we stored our items in the previous module) and just read it out from here?

Reply:
Changed to session item